ambry-frontend/src/main/java/com/github/ambry/frontend/FrontendRestRequestServiceFactory.java

```diff
@@ -94,23 +93,21 @@ public RestRequestService getRestRequestService() {
               clusterMap.getMetricRegistry()).getUrlSigningService();
       AccountAndContainerInjector accountAndContainerInjector =
           new AccountAndContainerInjector(accountService, frontendMetrics, frontendConfig);
-      StorageQuotaService storageQuotaService = null;
       AccountStatsStore accountStatsStore =
           Utils.<AccountStatsStoreFactory>getObj(frontendConfig.accountStatsStoreFactory, verifiableProperties,
               clusterMapConfig, new StatsManagerConfig(verifiableProperties),
               clusterMap.getMetricRegistry()).getAccountStatsStore();
-      if (frontendConfig.enableStorageQuotaService) {
-        storageQuotaService =
-            Utils.<StorageQuotaServiceFactory>getObj(frontendConfig.storageQuotaServiceFactory, verifiableProperties,
-                accountStatsStore, clusterMap.getMetricRegistry()).getStorageQuotaService();
-      }
+      QuotaConfig quotaConfig = new QuotaConfig(verifiableProperties);
```

Comment:
My concern here is that we are passing quotaManager object to SecurityService without calling quotaManager.init. For now it could be ok, but in general this is an unsafe practice, because there is nothing that prevents SecurityService from using uninitialized QuotaManager before the quotaManager.init is called in FrontendRestRequestService.
Why can't we call storageService.start() here?

Reply:
I am adding all the initialization work in the start method so the constructor method would be a light-weight method with only variable assignment statements. 
If we follow this rule, then the security service would not do anything in the its constructor.